src/test/java/org/broadinstitute/hellbender/tools/walkers/variantutils/SelectVariantsIntegrationTest.java

```diff
@@ -488,22 +495,241 @@ private void testFileWithoutInfoLineInHeader(final String name, final Class<? ex
+                        "-O", testOutputGenotype.getAbsolutePath(),
+                        "-select-genotype","DP > 20"),
+                SelectVariants.class.getSimpleName());
+        final Pair<VCFHeader, List<VariantContext>> resultGenotype = VariantContextTestUtils.readEntireVCFIntoMemory(testOutputGenotype.getAbsolutePath());
+        final int actualNumVCsGenotype = resultGenotype.getRight().size();
+        Assert.assertEquals(expectedNumVCsGenotype, actualNumVCsGenotype);
+    }
+
+    @Test
+    public void testCombineInfoAndGenotypeJexl() {
+        final File testOutput = createTempFile("jexl_genotype_info_test", "vcf");
+        beforeJexlTest();
+        final int gqThreshold = 60;
+        final Predicate<VariantContext> filter = vc -> vc.getAttributeAsString("SVTYPE", "").equals("DEL") ||
+                vc.getGenotypes().stream().anyMatch(g -> g.getGQ() > gqThreshold);
+        final Pair<VCFHeader, List<VariantContext>> preFilter = VariantContextTestUtils.readEntireVCFIntoMemory(svHGDBMultiSampleVcf.toString());
+        final int expectedNumVCs = (int) preFilter.getRight().stream().filter(filter).count();
+
+        runCommandLine(Arrays.asList("-V", svHGDBMultiSampleVcf.toString(),
+                        "-O", testOutput.getAbsolutePath(),
+                        "-" + SelectVariants.SELECT_SHORT_NAME,"SVTYPE == 'DEL'",
+                        "-" + SelectVariants.GENOTYPE_SELECT_SHORT_NAME,"GQ > " + gqThreshold),
+                SelectVariants.class.getSimpleName());
+
+        final Pair<VCFHeader, List<VariantContext>> result = VariantContextTestUtils.readEntireVCFIntoMemory(testOutput.getAbsolutePath());
+        final int actualNumVCs = result.getRight().size();
+        Assert.assertEquals(expectedNumVCs, actualNumVCs);
+
+        // Test the cases where we give an INFO field expression to --select-genotype, and vice versa.
+        // This works (unfortunately), since we are giving the VariantContext object to JEXL parser (so it can find info fields)
+        final Predicate<VariantContext> filter2 = vc -> vc.getAttributeAsString("SVTYPE", "").equals("DEL");
+        final int expectedNumVCs2 = (int) preFilter.getRight().stream().filter(filter2).count();
+        final File testOutput2 = createTempFile("jexl_genotype_info_mixup_test", "vcf");
+        runCommandLine(Arrays.asList("-V", svHGDBMultiSampleVcf.toString(),
+                        "-O", testOutput2.getAbsolutePath(),
+                        "-" + SelectVariants.GENOTYPE_SELECT_SHORT_NAME,"SVTYPE == 'DEL'"), // INFO filter expression given to genotype filter
+                SelectVariants.class.getSimpleName());
+        final Pair<VCFHeader, List<VariantContext>> result2 = VariantContextTestUtils.readEntireVCFIntoMemory(testOutput2.getAbsolutePath());
+        final int actualNumVCs2 = result2.getRight().size();
+        Assert.assertEquals(expectedNumVCs2, actualNumVCs2);
```

Comment:
This second test case should probably be a separate test method rather than embedded within `testCombineInfoAndGenotypeJexl()`

Reply:
Done